day2/exercise3/solution.go

```diff
@@ -0,0 +1,42 @@
+package main
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"sync"
+)
+
+var balance = 500
+
+func deposit(amount int, wg *sync.WaitGroup, m *sync.Mutex) {
+	m.Lock()
+	balance += amount
+	fmt.Println("Successfully deposit performed, current account balance", balance)
+	m.Unlock()
+	wg.Done()
+}
+
+func withdrawal(amount int, wg *sync.WaitGroup, m *sync.Mutex) {
+	m.Lock()
+	if balance >= amount{
+		balance -= amount
+		fmt.Println("Successfully withdrawal performed, current account balance", balance)
+	} else{
+		log.Fatal("lack of balance in account...")
+	}
+
+	m.Unlock()
+	wg.Done()
+}
+
+func main() {
+	var w sync.WaitGroup
+	var m sync.Mutex
```

Comment:
make it as a global variable as we are using in above func also, then it becomes simple

Reply:
done